README.md

```diff
@@ -271,21 +271,39 @@ public function getFoo() : string {
 ```
 
 ```php
-$this->database->executeGeneric("common.mistake.asynchronous", [], function() : void {
+$this->database->executeGeneric("beware.of.race_condition", [], function() : void {
 	$this->setFoo();
 });
 echo $this->getFoo();
 ```
-The result will be `bar` because the queries are run asynchronously. The code on the main thread will run faster than it.
+The result will be `bar` because the queries are run asynchronously. The code on the main thread will run earlier than it.
 
 To make the code give a correct result, you have to ensure `$this->setFoo()` runs before `echo $this->getFoo()`. The appropriate way is to move `getFoo()` into the callback function, just like below:
 ```php
-$this->database->executeGeneric("common.mistake.asynchronous", [], function() : void {
+$this->database->executeGeneric("beware.of.race_condition", [], function() : void {
 	$this->setFoo();
 	echo $this->getFoo();
 });
 ```
 
+### Returning the result from callback
+Due to race condition (as explained in the section above), it is not viable to return the result nor use it beyond the scope of the callback. If you are aiming to create an API function or seek to share the results with other parts of the code, it is advisable to convert the code into callbacks as well:
+
+```php
+public function myAPI(\Closure $userCallback)
+    $this->database->executeSelect("beware.of.return_from_callback", [], function($result) use ($userCallback) : void {
+	    $userCallback($result);
+    });
+
+    // Simpler versions:
+    $this->database->executeSelect("beware.of.return_from_callback", [], fn($result) => $userCallback($result));
+    $this->database->executeSelect("beware.of.return_from_callback", [], $userCallback);
+}
+```
+
+### Callbacks will clutter your code
+While using callbacks might be one straightforward solution to your problems, there exists a large trade off—you must sacrifice the code readability.
+Hence, [async/await](https://github.com/SOF3/await-generator) would also be a worthy topic which contributes your programming skillset despite its steep learning curve. At the end of the day, you will finally be able to overcome race condition and "return" the result!
```

Comment:
Is this a bit too subjective? "Worthy" and "your programming skillset" sound quite unprofessional here, and "steep learning curve" isn't exactly true since it's supposed to make things *easier* instead. And you cannot really "overcome" race conditions – race conditions still happen, they just don't appear so frequently (from 100% of your code to maybe 10% of your code, but you still have to deal with it sometimes).

Reply:
this why my eng fell from 5 to 3
paper three moment